sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/resolution/ExpressionResolver.kt

```diff
@@ -157,11 +161,19 @@ private fun Resolver.resolveFunction(
       }
       "abs", "coalesce", "ifnull", "likelihood", "likely", "nullif", "unlikely" -> {
         // Functions which return the type of their first argument.
+        if (resolutions[0].values.isEmpty()) {
+          return Resolution(errors = resolutions.flatMap { it.errors })
+        }
         sqliteType = resolutions[0].values[0].type
       }
       "max" -> {
         // NULL < INTEGER < REAL < TEXT < BLOB
-        resolutions.map { it.values[0].type }.forEach {
+        resolutions.map {
+          if (it.values.isEmpty()) {
+            return Resolution(errors = resolutions.flatMap { it.errors })
```

Comment:
It's not very functional to have these side-effects inside of a map. How about just the simple thing here:
```kotlin
if (resolutions.values.isEmpty()) {
  return //...
}
resolutions.values[0].type.forEach {
  // ...
```